### PR TITLE
add some more dump of jenkins pod logs on e2e test failures

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @adambkaplan  

(our newly minted owner :-) )

@openshift/sig-developer-experience fyi

### CODE REVIEW

**Comment by gabemontero** [comment]
aws quota limits for e2e-aws test run

File: test/extended/builds/pipeline_origin_bld.go [hunk 1]
```diff
@@ -385,24 +391,36 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
 				// NOTE, for the credential URL in Jenkins
 				// it returns rc 200 with no output if credential exists and a 404 if it does not exists
 				_, err = j.WaitForContent("", 200, 10*time.Second, "credentials/store/system/domain/_/credential/%s-%s/", oc.Namespace(), secretName)
+				if err != nil {
+					exutil.DumpApplicationPodLogs("jenkins", oc)
+				}
```
Comment (adambkaplan) [inline test/extended/builds/pipeline_origin_bld.go]: Why not add this to the `AfterEach` block at the start of the suite? See https://github.com/openshift/origin/blob/7d8731afc7213cda40cdfe0741cbf7723472a1a3/test/extended/builds/pipeline_origin_bld.go#L63-L67

Comment (gabemontero) [reply test/extended/builds/pipeline_origin_bld.go]: presumably the `exutil.DumpPodLogsStartingWith("", oc) ` should have caught it 

i suspect the pod is getting deleted now before we hit this path

Comment (adambkaplan) [reply test/extended/builds/pipeline_origin_bld.go]: @gabemontero looks like this test isn't using the Ginkgo `BeforeEach`/`AfterEach` conventions. How hard would it be to refactor the `cleanup` and `debugAnyJenkinsFailure` methods to this?

**Comment by gabemontero** [comment]
On Wed, Mar 13, 2019 at 3:46 PM Adam Kaplan <notifications@github.com>
wrote:

> *@adambkaplan* commented on this pull request.
> ------------------------------
>
> In test/extended/builds/pipeline_origin_bld.go
> <https://github.com/openshift/origin/pull/22307#discussion_r265302419>:
>
> > @@ -385,24 +391,36 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
>  				// NOTE, for the credential URL in Jenkins
>  				// it returns rc 200 with no output if credential exists and a 404 if it does not exists
>  				_, err = j.WaitForContent("", 200, 10*time.Second, "credentials/store/system/domain/_/credential/%s-%s/", oc.Namespace(), secretName)
> +				if err != nil {
> +					exutil.DumpApplicationPodLogs("jenkins", oc)
> +				}
>
> @gabemontero <https://github.com/gabemontero> looks like this test isn't
> using the Ginkgo BeforeEach/AfterEach conventions. How hard would it be
> to refactor the cleanup and debugAnyJenkinsFailure methods to this?
>
non-trivial refactor, and in fact we used to do this, but given the
slowness of and bloat wrt starting jenkins, we've bypassed normal ginkgo
conventions and coupled multiple tests together

we revert back and these tests run long and we have Clayton bugging us again

let's minimize churn at this point

**Comment by adambkaplan** [comment]
@gabemontero I'd like to see a fail case where we dump the Jenkins logs, then. Add a commit to force an error (which you can then remove).

/approve

**Comment by gabemontero** [comment]
sabotage pushed @adambkaplan

**Comment by gabemontero** [comment]
The sabotage was successful @adambkaplan 

Job https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/22307/pull-ci-openshift-origin-master-e2e-aws-builds/1027 dumped the jenkins pod on the failure as expected

Look at the stdout/stderr details for `openshift-tests [Feature:Builds][Slow] openshift pipeline build Sync plugin tests using the ephemeral template [Suite:openshift]`

**Comment by adambkaplan** [comment]
@gabemontero great! Remove the temp commit and I'll lgtm

**Comment by gabemontero** [comment]
temp commit removed @adambkaplan  ... bump

**Comment by gabemontero** [comment]
aws network resource quota pain on the e2e-aws and e2e-aws-serial failures an hour ago

**Comment by gabemontero** [comment]
apparent api.ci hiccup

**Pull request merged**